snake.py

```diff
@@ -0,0 +1,227 @@
+import pygame, sys
+from pygame import *
+import random
+
+class Player(object):
+    '''
+    Player's snake object handling head position and direction, body part position and order, and moving and drawing its parts.
+    '''
+
+    def __init__(self,pos,direction):
+
+        self.pos = pos
+        self.direction = direction
+        self.size = 20
+        self.body = []
+
+        ### Colors
+        PURPLE = (128,0,128)
```

Comment:
Nice that you are capitalizing the constants. One nice way to organize these values is to create a file class named Config.py and initialize it as an object before running the main loop of the game.

Reply:
Also, constants (and other variables that don't depend on the specific instance) can go outside the method definition, in the class or in the module (top level of the file), even if they don't go in a separate `constants.py`.

Typically, since this is a constant:

```python

DARK_BLUE = (0, 0, 204)

class Player(object):
    …
    def __init__(self, pos, direction):
        …
        self.head_color = DARK_BLUE
````

If the value is specific to this particular class (not true here), you could also write:

```python

class Player(object):
    DARK_BLUE = (0, 0, 204)
    …
    def __init__(self, pos, direction):
        …
        self.head_color = DARK_BLUE
````